### PR TITLE
Changes and fixes to prepare for dynamic conv

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68176
* __->__ #68175

Summary: This slightly alters the way from_float works so it will work
with placeholder observers. It also fixes a but with ConvTranspose3d and
ConvTranspose1d where the parameters like kernel_size, stride...etc
weren't set properly. New tests were added to check for this type of
issue as well.

Test Plan: python test/test_quantization.py TestQuantizedOps
python test/test_quantization.py TestStaticQuantizedModule

Reviewers:

Subscribers:

Tasks:

Tags:

ALL dynamic conv fixes

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32374004](https://our.internmc.facebook.com/intern/diff/D32374004)

### CODE REVIEW

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32374004).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32374004).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32374004).

File: torch/nn/quantized/modules/conv.py [hunk 1]
```diff
@@ -637,11 +647,11 @@ def __init__(self, in_channels, out_channels, kernel_size, stride=1,
...
-        kernel_size = _pair(kernel_size)
-        stride = _pair(stride)
-        padding = _pair(padding)
-        dilation = _pair(dilation)
-        output_padding = _pair(output_padding)
+        kernel_size = _single(kernel_size)
+        stride = _single(stride)
+        padding = _single(padding)
+        dilation = _single(dilation)
+        output_padding = _single(output_padding)
```
Comment (z-a-f) [inline torch/nn/quantized/modules/conv.py:654]: Please, double check that -- I remember long ago the conv1d was just a special case for the conv2d. If so, we need a pair here.

**Review by z-a-f — APPROVED** [review]

File: torch/nn/quantized/modules/conv.py (hunk 1 above)
Comment (HDCharles) [reply torch/nn/quantized/modules/conv.py:654]: yeah I'll add a run to the test

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32374004).

File: torch/nn/quantized/modules/conv.py [hunk 2]
```diff
@@ -204,6 +203,12 @@ def get_qconv(cls, mod, activation_post_process, weight_post_process=None):
                     mod.stride, mod.padding, mod.dilation, mod.groups,
                     mod.bias is not None, mod.padding_mode)
         qconv.set_weight_bias(qweight, mod.bias)
+        try:
+            act_scale, act_zp = activation_post_process.calculate_qparams()
+        except Exception:
+            # for things like dynamic quantization, want to assign arbitrary s+z to not break serialization
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/conv.py:209]: why do we call this from_float for dynamically quantized conv module? shouldn't we have a nn.quantized.dynamic.Conv2d?

Comment (HDCharles) [reply torch/nn/quantized/modules/conv.py:209]: sure but the functions are exactly the same aside from dynamic not needing params

**Pull request closed without merging**